controllers/base_controller.js

```diff
@@ -115,8 +115,10 @@ module.exports = function BaseControllerModule(pb) {
         this.templateService.registerLocal('localization_script', function(flag, cb) {
             self.requiresClientLocalizationCallback(flag, cb);
         });
-        this.templateService.registerLocal('analytics', function(flag, cb) {
-            pb.AnalyticsManager.onPageRender(self.req, self.session, self.ls, cb);
+        pb.AnalyticsManager.onPageRender(self.req, self.session, self.ls, function(err, results){
+          results.forEach(function(result){
+            self.templateService.registerLocal(result.tag, new pb.TemplateValue(result.snippet, false));
+          });
```

Comment:
I don't think this is the right approach here for two reasons:

1) There is a performance concern because calling the onPageRender every time a base controller is instantiated is expensive and you don't know if it is a view or API controller so it could potentially be wasted resources.
2) You execute asynchronously  but do not wait so there is no guarantee that the onPageRender will be available by the time the page renders.

Potential solution:
I think you would be better off to provide a function that returns a list of tags and then register each tag.  This way you would only call the individual analytic providers if and only if they are encountered in the template.  You could then pass the specific flag/tag to the ```pb.AnalyticsManager.onPageRender``` function to know which specific provider to render.